extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -427,7 +435,15 @@ protected static Uni<JsonWebKeySet> getJsonWebSetUni(OidcProviderClient client,
                     .transform(new Function<Throwable, Throwable>() {
                         @Override
                         public Throwable apply(Throwable t) {
-                            return toOidcException(t, oidcConfig.authServerUrl.get());
+                            return toOidcException(t, oidcConfig.authServerUrl.get(),
+                                    oidcConfig.tenantId.orElse(DEFAULT_TENANT_ID));
+                        }
+                    })
+                    .invoke(new Runnable() {
```

Comment:
Is it possible to make it conditional ? Creating a runnable instance if no one is listening will have some cost

Reply:
done